File: src/my_project_typescript/easy_problems_typescript/from101to150/reformatPhoneNumber.ts
function reformatNumber(number: string): string {

    let reOne = / /gi
    let reTwo = /-/gi
    let temp: string = number.replace(reOne, "") 
    temp = temp.replace(reTwo, '')

    let answer: Array<string> = []
    let count: number = 0

    while (count < temp.length) {
        if (temp.length - count == 4){
            answer.push(temp.slice(count, count+2))
            answer.push(temp.slice(count+2, count+4))
            break
        }
        if (temp.length - count == 3){
            answer.push(temp.slice(count, count+3))
            break
        }
        if (temp.length - count == 2){
            answer.push(temp.slice(count, count+2))
            break
        }
        answer.push(temp.slice(count, count+3))
        count += 3
    }

    return answer.join('-')
};

console.log(reformatNumber("1-23-45 6"))

console.log(reformatNumber("123 4-567"))

console.log(reformatNumber("123 4-5678"))

console.log(reformatNumber("12"))

console.log(reformatNumber("--17-5 229 35-39475 "))

File: src/my_project/easy_problems/from201to250/reformat_phone_number.py
from typing import List 

class Solution:
    def reformatNumber(self, number: str) -> str:
        
        
        temp = number.replace(' ','').replace('-','')
        
        answer = []
        
        count = 0
        while count < len(temp):
            if len(temp)-count == 4:
                answer.append(temp[count:count+2])
                answer.append(temp[count+2:count+4])
                break 
            if len(temp) - count  == 3:
                answer.append(temp[count: count+3])
                break 
            if len(temp) - count  == 2:
                answer.append(temp[count:count+2])
                break 
            answer.append(temp[count:count+3])
            count += 3
        
        return '-'.join(answer)
    
    
solution = Solution()

print(solution.reformatNumber(number = "1-23-45 6"))

print(solution.reformatNumber(number = "123 4-567"))

print(solution.reformatNumber(number = "123 4-5678"))

print(solution.reformatNumber(number = "12"))

print(solution.reformatNumber(number = "--17-5 229 35-39475 "))